src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -496,8 +517,50 @@ public TStructuralType Patch(TStructuralType original)
         /// <summary>
         /// Overwrites the <paramref name="original"/> entity with the values stored in this delta.
         /// </summary>
+        /// <param name="original">The entity to be updated.</param>
+        /// <param name="apiHandler">API Handler for the entity.</param>
+        /// <param name="apiHandlerFactory">API Handler Factory.</param>
+        public void Patch(TStructuralType original, IODataAPIHandler apiHandler, ODataAPIHandlerFactory apiHandlerFactory)
+        {
+            Debug.Assert(apiHandler != null, "apiHandler != null");
+
+            if (IsComplexType)
+            {
+                original = ReassignComplexDerivedType(original, _structuredType, original.GetType(), ExpectedClrType) as TStructuralType;
+            }
+
+            CopyChangedValues(original, apiHandler as ODataAPIHandler<TStructuralType>, apiHandlerFactory);            
+        }
+
+        /// <summary>
+        /// This is a patch on ODataId. It applies the ODataId parsed navigation paths, gets the identified values and copies them to the original object.
+        /// </summary>    
```

Comment:
Could you also add comments for the parameters?

Reply:
Refactored some logic and removed this method